watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination_withRO.py

```diff
@@ -0,0 +1,568 @@
+
+    # cost nanofiltration module and pump
+    prtrt.wwtp.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.zo_costing}
+    )
+    dye_sep.nanofiltration.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.zo_costing}
+    )
+    dye_sep.P1.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.zo_costing}
+    )
+
+    # RO Train
+    # RO equipment is costed using more detailed costing package
+    desal.P2.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.ro_costing,
+            "costing_method_arguments": {"cost_electricity_flow": False},
+        }
+    )
+    desal.RO.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.ro_costing}
+    )
+
+    desal.M1.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.ro_costing}
+    )
+    desal.PXR.costing = UnitModelCostingBlock(
+        default={"flowsheet_costing_block": m.fs.ro_costing}
+    )
+    desal.P3.costing = UnitModelCostingBlock(
+        default={
+            "flowsheet_costing_block": m.fs.ro_costing,
+            "costing_method_arguments": {"cost_electricity_flow": False},
+        }
+    )
+
+    # Aggregate unit level costs and calculate overall process costs
+    m.fs.zo_costing.cost_process()
+    m.fs.ro_costing.cost_process()
+
+    # Annual disposal of brine
+    m.fs.brine_disposal_cost = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * (
+                m.fs.zo_costing.waste_disposal_cost
+                * pyunits.convert(
+                    m.fs.brine.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            )
+        ),
+        doc="Cost of disposing of brine waste",
+    )
+
+    m.fs.dye_recovery_revenue = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * m.fs.zo_costing.dye_mass_cost
+            * pyunits.convert(
+                m.fs.dye_retentate.flow_mass_comp[0, "dye"],
+                to_units=pyunits.kg / m.fs.zo_costing.base_period,
+            )
+        ),
+        doc="Savings from dye recovered back to the plant",
+    )
+
+    m.fs.water_recovery_revenue = Expression(
+        expr=(
+            m.fs.zo_costing.utilization_factor
+            * m.fs.zo_costing.recovered_water_cost
+            * pyunits.convert(
+                m.fs.permeate.properties[0].flow_vol,
+                to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+            )
+        ),
+        doc="Savings from water recovered back to the plant",
+    )
+
+    # Combine results from costing packages and calculate overall metrics
+    @m.Expression()
+    def total_capital_cost(b):
+        return (
+            pyunits.convert(
+                m.fs.zo_costing.total_capital_cost, to_units=pyunits.USD_2018
```

Comment:
Why convert to 2018 instead of keeping at 2020? In that case, I suppose you'd convert `ro_costing` to 2020. I may be wrong, but I thought we were working with USD_2020 for the AMO cases.

Reply:
I believe some of the other examples (metab) were in 2018 so I went with that, but I can easily change to 2020.

Reply:
Ah ok- that's what I was failing to recall. In that case, it might make sense to use USD_2018 in the flowsheet's global param YAML file